test/dom/dom.ts

```diff
@@ -0,0 +1,493 @@
+      let l: any = ion.load('planets::["Mercury", "Venus", "Earth", "Mars"]')!;
+
+      assert.equal(IonTypes.LIST, l.getType());
+      assert.deepEqual(['planets'], l.getAnnotations());
+
+      // Iteration
+      for (let planet of l) {
+         assert.isTrue(planet instanceof String);
+      }
+
+      // Indexing
+      assert.equal("Mercury", l[0]);
+      assert.equal("Venus", l[1]);
+      assert.equal("Earth", l[2]);
+      assert.equal("Mars", l[3]);
+
+      assert.equal(4, l.length);
+
+      // Directly use Array methods
+      assert.isTrue(l.findIndex(s => s == "Earth") > 0);
+      assert.isNotNull(l.find(s => s == "Earth"));
+   });
+
+   it('load() SExpression as Value', () => {
+      let s: Value = ion.load('planets::("Mercury" "Venus" "Earth" "Mars")')!;
+
+      assert.equal(IonTypes.SEXP, s.getType());
+      assert.deepEqual(['planets'], s.getAnnotations());
+
+      // Iteration
+      for (let planet of s.values()!) {
+         assert.isNotNull(planet.stringValue());
+      }
+
+      // Indexing
+      assert.equal("Mercury", s.get(0)!.stringValue());
+      assert.equal("Venus", s.get(1)!.stringValue());
+      assert.equal("Earth",s.get(2)!.stringValue());
+      assert.equal("Mars", s.get(3)!.stringValue());
+
+      let planets: string[] = [...s.values()!].map(s => s.stringValue()!);
+      assert.equal(4, planets.length);
+   });
+
+   it('load() SExpression as any', () => {
+      let s: any = ion.load('planets::("Mercury" "Venus" "Earth" "Mars")')!;
+
+      assert.equal(IonTypes.SEXP, s.getType());
+      assert.deepEqual(['planets'], s.getAnnotations());
+
+      // Iteration
+      for (let planet of s) {
+         assert.isTrue(planet instanceof String);
+      }
+
+      // Indexing
+      assert.equal("Mercury", s[0]);
+      assert.equal("Venus", s[1]);
+      assert.equal("Earth", s[2]);
+      assert.equal("Mars", s[3]);
+
+      assert.equal(4, s.length);
+
+      // Directly use Array methods
+      assert.isTrue(s.findIndex(s => s == "Earth") > 0);
+      assert.isNotNull(s.find(s => s == "Earth"));
+   });
+
+   it('load() Struct as Value', () => {
+      let s: Value = ion.load(
+          'foo::bar::{' +
+            'name: {' +
+               'first: "John", ' +
+               'middle: "Jacob", ' +
+               'last: "Jingleheimer-Schmidt",' +
+            '},' +
+            'age: 41' +
+          '}'
+      )!;
+
+      assert.equal(IonTypes.STRUCT, s.getType());
+      assert.deepEqual(['foo', 'bar'], s.getAnnotations());
+
+      // Field access
+      assert.equal("Jacob", s.get("name")!.get("middle")!.stringValue());
+      assert.equal("Jacob", s.get("name", "middle")!.stringValue());
+
+      // Iteration
+      for (let [fieldName, value] of s.fields()!) {
+         assert.isTrue(typeof fieldName === "string");
+         assert.isTrue(fieldName.length > 0);
+         assert.isFalse(value.isNull());
+      }
+
+      assert.equal(2, [...s.fields()!].length)
```

Comment:
`[...s.fields()!]` --> `s.fields()!` ?

Reply:
`s.fields()` returns an iterator and so does not have a `length` property.